### ISSUE TITLE
[Feature] Add phpStan stub file so 

### ISSUE THREAD

**Issue by NickStallman** [DataDog/dd-trace-php#1574]
**Is your feature request related to a problem? Please describe.**
phpstan will statically analyse source code to find potential issues. Currently as ddtrace defines functions like trace_method and classes like SpanData that phpstan doesn't know about, it will complain loudly about any custom tracing code.

**Describe the solution you'd like**
An official stub file to give phpStan the information it needs would be fantastic.

An initial file I've been using is: ddtrace.stub
```php
<?php

namespace DDTrace;

/**
 * @param array<mixed> $args
 */
function trace_method(\DDTrace\SpanData $span, array $args, mixed $retval, ?\Exception $exception): void
{
}

namespace DDTrace;
class SpanData
{
    public string $name;
    public string $resource;
    public string $service;
    public string $type;
    /** @var string[] $meta */
    public $meta;
    /** @var float[] $metrics */
    public $metrics;
}
```

**Comment by bwoebi** [comment]
We have planned on moving towards PHP stub files (As a starting point I've begun experimenting with https://github.com/DataDog/dd-trace-php/blob/52106acf64fb45acd6c74f85fc58adead4a5ff4c/ext/hook/uhook.stub.php).

Expect to see this becoming reality in the coming months :-)

**Comment by ralphschindler** [comment]
Looking forward to stub files, any movement here? Thanks!

**Comment by bwoebi** [comment]
We are actively planning on doing this in the next 2-3 months.

**Comment by bwoebi** [comment]
We now have the following .stub.php files (to be released soon as 0.85.0):

- ext/ddtrace.stub.php
- ext/hook/uhook.stub.php
- ext/hook/uhook_attributes.stub.php

@ralphschindler and @NickStallman I hope that's solving all the doc and autocompletion issues you were having :-)

**Comment by NickStallman** [comment]
Brilliant! Thanks. :)
